jigsaw/client/menu.py

```diff
@@ -0,0 +1,167 @@
+import random
+
+import pygame_gui
+from pygame import init, draw, Rect, display, time, event, quit
+from pygame.locals import QUIT, USEREVENT
+
+from client.constants import WHITE, BLUE, WIDTH, HEIGHT, screen, font, manager
+from client.credits_view import CreditsView
+from client.settings_view import SettingsView
+
+init()
+
+'''
+# Colors
+
+WHITE = (255, 255, 255)
+RED = (255, 0, 0)
+BLUE = (0, 0, 255)
+
+# Screen dimensions
+WIDTH, HEIGHT = 800, 600
+screen = pygame.display.set_mode((WIDTH, HEIGHT))
+font = pygame.font.SysFont(None, 36)  # Default font, size 36
+overlap_sensitivity = 5
+
+pygame.display.set_caption('Game Menu')
+
+'''
+
+
+class PlayGameView:
+    pass
+
+class Menu:
+    def __init__(self):
+        self.view_instances = {
+            "settings": SettingsView(),
+            "credits": CreditsView(),
+            "play_game": PlayGameView(),
+        }
+        self.active_view = self
+        self.clock = time.Clock()
+        self.build()
+
+    def build(self):
+        button_size = (100, 50)
+        right_edge = WIDTH - button_size[0]
+        count = 83
+        MODES = ["quick", "secret", "settings", "credits"]
+        for mode in MODES:
+            rectangle = Rect((right_edge - 50, count), button_size)
+            button = pygame_gui.elements.UIButton(
+                relative_rect=rectangle, text=mode.title(), manager=manager)
+            count += 83
+            setattr(self, f"{mode}_button", button)
+
+        self.right_panel = pygame_gui.elements.UIPanel(
+            relative_rect=Rect((WIDTH - 200, 0), (200, HEIGHT)),
+            starting_height=0,
+            manager=manager
+        )
+
+    def draw_completion_bar(self, completion_percent, position, size):
+        """ Draws a completion bar on the screen.
+        Args:
```

Comment:
```suggestion
    def draw_completion_bar(self, completion_percent: float, position: tuple[int, int], size: tuple[int, int]) -> None:
        """ Draws a completion bar on the screen.
        
        Args:
```

Reply:
this is a game client thing, for when someones solving the puzzle move it over there